Add option to disable EQ gain influence on waveforms

artiniest opened the pull request:
![image](https://github.com/user-attachments/assets/9754967d-45da-4d30-a1a5-40e82aebc748)

Adds a new option in Preferences > Waveforms to disable the visible effect of EQ gain knobs on waveform rendering. Visual gain remains applied independently.

[Feature request here](https://github.com/mixxxdj/mixxx/issues/14901)

artiniest committed:
DlgPrefWaveform: add setting to disable EQ gain influence on waveform

Adds a checkbox to preferences to toggle if EQ knob gain affects
waveform rendering. Allows displaying waveform amplitude regardless
of EQ knob positions.

artiniest committed:
Merge branch 'mixxxdj:main' into eq-waveform

Swiftb0y commented:
Welcome and thanks for improving this! Please sign the [Mixxx Contributor Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ) and comment here when you have done so. It gives us permission to distribute your contribution under the GPL v2 or later license and the Apple Mac App Store. It is also helpful for us to have contact information for contributors in case we may need it in the future.

artiniest commented:
I've signed the agreement.

Swiftb0y reviewed (commented):
this looks very decent already. I have a couple requests:

artiniest committed:
DlgPrefWaveform: rename and fix EQ waveform visualization option

Flipped the setting logic to "visualize EQ waveform".
Fixed a logic issue where dead code was not skipped.

artiniest commented:
@Swiftb0y Addressed the points you brought up! Let me know if you notice anything else that I should fix.

Swiftb0y commented on src/preferences/dialog/dlgprefwaveform.cpp:
```diff
@@ -284,9 +284,9 @@ void DlgPrefWaveform::slotUpdate() {
-    bool disableEqGain = m_pConfig->getValue(
-            ConfigKey("[Waveform]", "disable_eq_gain"), false);
-    disableEqGainCheckBox->setChecked(disableEqGain);
+    bool visualizeEqGain = m_pConfig->getValue(
+            ConfigKey("[Waveform]", "visualize_eq_gain"), false);
```
lets default this to true since that was the default previously.
```suggestion
    bool visualizeEqGain = m_pConfig->getValue(
            ConfigKey("[Waveform]", "visualize_eq_gain"), true);
```

Swiftb0y reviewed (commented):
Thanks. Last nit I think.

Swiftb0y commented:
I mean there is the general issue that this setting only applies to some waveform types (RGB, Stacked) and not others, but all of those settings are shown regardless of whether they have an effect on the current type or not. Addressing this is not a requirement for this PR though (you're welcome to look into it in a follow up PR if you'd like to).

artiniest committed:
DlgPrefWaveform: default EQ visualization option to true

Swiftb0y reviewed (commented):
thoughts @acolombier?

acolombier commented:
I think it would be best to rely on [Waveform option](https://github.com/mixxxdj/mixxx/blob/38a400a0571a041b7b68055dcf55a38ff7d6ad4c/src/waveform/renderers/allshader/waveformrenderersignalbase.h#L18) rather than adding a context-less setting. This will allows a better integration with the cleanup work we've made with @m0dB , prevent growing more config key and ease porting on QML/Mixxx 3.0 as not building on the WaveformFactory/dedicated config key.

acolombier commented:
Indeed - I'm wondering if there is a middle ground we could reach, just to make sure we keep things clean. Let me see if I can move the option outside `allshader`

acolombier commented:
Hi @artiniest, waveform options should now be available outside of `allshader` waveform!

DidierMalenfant commented:
> Please also note the feedback for RGB types is broken. Here is the fix: #15808

Ah yes... that explains it then. I merged that PR on top of mine and everything is looking good now.

One thing I noticed is that the original PR for this retains a test that keeps getting removed and added back with various merges in main. In `waveformrenderersignalbase.cpp`:

```
if (m_pEQEnabled && m_pEQEnabled->get() > 0.0) {
```

The test for `m_pEQEnabled` was removed in https://github.com/mixxxdj/mixxx/commit/4cc1307b but then added back in a merge from 2.6 at https://github.com/mixxxdj/mixxx/commit/d628d80f. I just want to make sure this can stay in with this PR.

loco-r commented:
bonjour 
avez-vous réussi à ajouter cette option très utile ?
